compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -364,17 +364,13 @@ where
             // happens at run-time so that's okay.
             let align = match self.size_and_align_of(&base.meta, &field_layout)? {
                 Some((_, align)) => align,
-                None if offset == Size::ZERO => {
-                    // An extern type at offset 0, we fall back to its static alignment.
+                None => {
+                    // An extern type, we fall back to its static alignment.
```

Comment:
I have no idea if this is correct. Would be good to have a link to the equivalent code in codegen to make sure at least both sides are consistent here (and maybe add cross-references to ensure they remain consistent).

Reply:
Oh, I thought I could just leave the existing fall back to the static alignment, presuming it would be effectively no-op, but that wouldn't be correct for packed representation. Fixed now.

The code generation implementation is in `PlaceRef::project_field`. It checks for the extra metadata first and when metadata is absent it performs no additional adjustments.